ykrt/src/compile/mod.rs

```diff
@@ -28,6 +27,50 @@ pub(crate) mod jitc_llvm;
 #[cfg(jitc_yk)]
 pub mod jitc_yk;
 
+/// A failure to compile a trace.
+#[derive(Debug, thiserror::Error)]
+pub enum CompilationError {
+    #[error("Unrecoverable error: {0}")]
+    Unrecoverable(String),
+    #[error("Temporary error: {0}")]
+    Temporary(TemporaryErrorKind),
```

Comment:
Do we gain anything useful from `TemporaryErrorKind`? My guess is that, at least at the moment, we will not find its contents valuable other than printing them out for debugging. Perhaps in most cases we even want a `temporary_compilation_error!()` macro which expands to `CompilationError::Temporary(format!("Line {}", line!())` or similar? It cuts down on us having to type stuff but means that when we're debugging we'll know *which* temporary error caused a problem.

Reply:
I think you are right. We don't actually care what's inside. I just copied precedence from [elsewhere](https://github.com/ykjit/yk/blob/7dc01b9ccc79bf629a2136cf545bc6db35c3ffd0/hwtracer/src/errors.rs#L15).

Maybe that should be updated at some point too (but not now).

Reply:
Fixed in https://github.com/ykjit/yk/pull/956/commits/ab849eb9d74e139a62c5466dca6f2c283162f1e5